pkg/controller/registry/reconciler/configmap.go

```diff
@@ -257,6 +258,20 @@ func (c *ConfigMapRegistryReconciler) EnsureRegistryServer(catalogSource *v1alph
 	if source.Spec.SourceType == v1alpha1.SourceTypeConfigmap || source.Spec.SourceType == v1alpha1.SourceTypeInternal {
 		// fetch configmap first, exit early if we can't find it
 		configMap, err := c.Lister.CoreV1().ConfigMapLister().ConfigMaps(source.GetNamespace()).Get(source.Spec.ConfigMap)
+		// Attempt to look up the CM via api call if there is a cache miss
+		if k8serrors.IsNotFound(err) {
```

Comment:
What happens if I write 

```yaml
spec:
  configMap: this-one-doesn't-exist-at-all
```

won't it hotloop and hit the cluster every time?

Reply:
If the configmap is missing, there will be a cache miss and an extra api call to the cluster and it fails out. We don't requeue the catsrc for configmap type so it shouldn't cause a hotloop here. This is the tradeoff between reducing cache size and extra API calls for some corner cases such as missing configmap and initial label updating.

Reply:
I think this is a reasonable tradeoff and Ben concurred. However, if you think it is not worth it, then I will drop this PR. Not a big deal.